src/Queue.cpp

```diff
@@ -50,6 +50,21 @@ void EventQueue::enqueue(
   const std::string &toDirectory,
   const std::string &toFile
 ) {
+  if (mPaused) {
+    return;
+  }
   std::lock_guard<std::mutex> lock(mutex);
   queue.emplace_back(std::unique_ptr<Event>(new Event(type, fromDirectory, fromFile, toDirectory, toFile)));
 }
+
+void EventQueue::pause() {
+  mPaused = true;
+}
+
+void EventQueue::resume() {
+  mPaused = false;
+}
```

Comment:
You need to have the check and set of each of these values behind a synchronization primitive or it's undefined behavior.

Reply:
You can refer to this article https://devblogs.microsoft.com/oldnewthing/20180924-00/?p=99805